### PR TITLE
--show-all is inert in v1.11

### PR DESCRIPTION
**Pull request by ChiuWang**
**What this PR does / why we need it**:

`--show-all` is inert in v1.11
ref: #60210

**Special notes for your reviewer**:
/assign @deads2k 

**Release note**:

```release-note
`--show-all` (which only affected pods and only for human readable/non-API printers) is inert in v1.11, and will be removed in a future release.
```

### CODE REVIEW

**Comment by thockin** [comment]
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/60793)

**Review by juanvallejo — APPROVED** [review]

**Comment by juanvallejo** [comment]
@soltysh this lgtm
@charrywanganthony thanks, looks like you'll also need to update `printers_test.go:1574`

**Comment by charrywanganthony** [comment]
@juanvallejo Modified. :smile:

File: pkg/kubectl/resource_filter.go
Comment (deads2k) [inline pkg/kubectl/resource_filter.go]: why is this still here?

Comment (charrywanganthony) [reply pkg/kubectl/resource_filter.go]: I thought it should be removed in the future. Should I remove it in this PR? @deads2k

Comment (deads2k) [reply pkg/kubectl/resource_filter.go]: The inert bit is about the flag which needs to continue to exist so someone specifying it doesn't get a failure.  We don't need to keep cruft in the codebase though.

Comment (charrywanganthony) [reply pkg/kubectl/resource_filter.go]: @deads2k Rebased.

File: pkg/kubectl/cmd/resource/get_test.go
Comment (deads2k) [inline pkg/kubectl/cmd/resource/get_test.go]: sorry I didn't notice before.  This entire test function is now dead.

File: pkg/kubectl/resource_filter.go
Comment (deads2k) [inline pkg/kubectl/resource_filter.go]: There are no filters anymore, so this type should be removed.

**Comment by deads2k** [comment]
/approve

@juanvallejo you have lgtm

File: pkg/kubectl/resource_filter.go
Comment (charrywanganthony) [reply pkg/kubectl/resource_filter.go]: And how about the entire `resource_filter.go` file ?

Comment (juanvallejo) [reply pkg/kubectl/resource_filter.go]: Since we only care about keeping the flag, I would be in favor of removing this file (and any references to it in factory_client_acces.go, get.go)

**Comment by jdumars** [comment]
Code freeze will lift on 3/19, and this will merge after that.

**Review by soltysh — APPROVED** [review]
/lgtm

**Comment by charrywanganthony** [comment]
It seems need lgtm again, and Milestone Pull Request Labels. @deads2k @juanvallejo @soltysh

**Review by soltysh — APPROVED** [review]
/lgtm

**Comment by charrywanganthony** [comment]
@juanvallejo still needs LGTM label. :smile:

**Review by juanvallejo — APPROVED** [review]

**Review by soltysh — APPROVED** [review]
/lgtm

**Pull request merged**
